mvc-api/mvc-service/app/managers/EventObjectProducer.java

```diff
@@ -0,0 +1,37 @@
+package managers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.json.simple.JSONObject;
+import org.sunbird.search.util.SearchConstants;
+
+import java.util.LinkedHashMap;
+import java.util.UUID;
+
+public class EventObjectProducer {
+    EventProducer eventProducer = new EventProducer();
+    public void addToEventObj(JSONObject content, String contentId)  {
+        // Add label
+        content.put("label","MVC");
+        ObjectMapper event = new ObjectMapper();
+        try {
+
+            JSONObject eventObj = event.readValue(SearchConstants.autocreatejobevent, JSONObject.class);
+            final long timeinmillisecond = System.currentTimeMillis();
+            eventObj.put("ets", timeinmillisecond);
+            UUID uniqueKey = UUID.randomUUID();
+            String mid = "VD." + timeinmillisecond + "." + uniqueKey;
+            eventObj.put("mid", mid);
+            LinkedHashMap<String,Object> object = (LinkedHashMap<String,Object>) eventObj.get("object");
+            object.put("id", contentId);
+            LinkedHashMap<String,Object> context = (LinkedHashMap<String,Object>) eventObj.get("context");
+            context.put("channel", content.get("channel").toString());
+            LinkedHashMap<String,Object> edata = (LinkedHashMap<String,Object>) eventObj.get("edata");
+            edata.put("repository", SearchConstants.vidyadaanurl + contentId);
+            edata.put("metadata", content);
+            eventProducer.writeToKafka(eventObj.toString(), SearchConstants.mvctopic);
+        }
+        catch (Exception e) {
+            System.out.println("Exception in Eventobjectproducer " + e);
```

Comment:
The exception is suppressed here. We should throw it to the above level and return SERVER_ERROR (HTTP 500) or CLIENT_ERROR (HTTP 400) based on the error type.